### PR TITLE
Docs 6 cleanup

### PR DESCRIPTION
**Pull request by stevepiercy**
This PR cleans up the unresolved suggestions from https://github.com/plone/plone.api/pull/469

See commit history for a detailed explanation.

### CODE REVIEW

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
Is it just me, or are there some missing checks and builds? I expected to see 5 from Netlify, but they do not show up. I wonder if it is because I am not able to create a new branch on `plone.api` and had to fork it and push from my remote, whereas @ksuess was able to create branch in this repo?

**Comment by stevepiercy** [comment]
All green. Yay! I would appreciate a review from another person, at their convenience.

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

**Comment by stevepiercy** [comment]
This PR also restores doctests and should fix https://github.com/plone/plone.api/issues/474, except for the 5 doctests that fail.

**Comment by stevepiercy** [comment]
I will need help from someone to fix the 5 failing doctests, preferably someone who is familiar with them and `plone.api`.

I also do not have permission to create a branch on this repo, but if granted access, I can close this PR and open another on a new branch in this repo.

Please let me know. Thank you!

**Comment by stevepiercy** [comment]
One more thing, these tests do not run (nor should they run) in the main `documentation` repo, but it will cause Sphinx errors. We (@ksuess and I) still need to figure something to handle that. I tried using the [`ifconfig` extension](https://www.sphinx-doc.org/en/master/usage/extensions/ifconfig.html), and that seems to work well.

## `conf.py`

```
extensions = [
    "sphinx.ext.ifconfig",
]

def setup(app):
    app.add_config_value("plone_api_doctests", "", True)  # False in `documentation`
```

## Example MyST

````
```{eval-rst}
.. ifconfig:: plone_api_doctests
.. invisible-code-block:: python

    self.assertEqual(portal.__class__.__name__, 'PloneSite')
    self.assertEqual(catalog.__class__.__name__, 'CatalogTool')
    self.assertEqual(user.__class__.__name__, 'MemberData')
```
````

That works in `plone.api` but not in `documentation`. I'll keep fiddling around with it.

File: docs/contribute/develop.md [hunk 1]
```diff
@@ -29,7 +31,7 @@ First let's look at 'system' libraries and applications that are normally instal
 - `GNU make` - The fundamental build-control tool.
 - `GNU tar` - The (un)archiving tool for extracting downloaded archives.
 - `bzip2` and `gzip` decompression packages - `gzip` is nearly standard, however some platforms will require that `bzip2` be installed.
-- `Python 2.7` - Linux distributions normally already have it, OS X users should use <https://github.com/collective/buildout.python> to get a clean Python version (the one that comes with OS X is broken).
+- `Python 3.9` - Linux distributions normally already have it, OS X users should use <https://github.com/collective/buildout.python> to get a clean Python version (the one that comes with OS X is broken).
```
Comment (jensens) [inline docs/contribute/develop.md:34]: I doubt `buildout.python` is a thing theses days. This needs an update.

File: docs/contribute/develop.md [hunk 2]
```diff
@@ -3,10 +3,12 @@
 This section is meant for contributors to the `plone.api` project.
 Its purpose is to guide them through the steps needed to start contributing.
 
-% note ``` This HowTo is written for Linux and OS X users.
-% If you're are running Windows we suggest using VMWare or a similar Virtualization tool to install Ubuntu Linux on a virtual machine or installing Ubuntu Linux as a secondary OS on your machine.
-% Alternatively, you can browse Plone's documentation on how to get Plone development environment up and running on Windows.
-% Plone does run on Windows but it's not completely trivial to set it up.
+```{note}
+This HowTo is written for Linux and OS X users.
+If you are running Windows, we suggest either using VMWare or a similar virtualization tool to install Ubuntu Linux on a virtual machine, or installing Ubuntu Linux as a secondary operating system on your machine.
```
Comment (jensens) [inline docs/contribute/develop.md:8]: The windows part should point to WSL. This needs an update.

File: docs/contribute/release.md [hunk 3]
```diff
@@ -13,8 +13,8 @@ Releasing a new version of `plone.api` involves the following steps:
 To avoid human errors and to automate some of the tasks above we use `jarn.mkrelease`.
 It's listed as a dependency in `setup.py` and should already be installed in your local bin:
 
-```bash
-$ bin/mkrelease --help
+```shell
+bin/mkrelease --help
```
Comment (jensens) [inline docs/contribute/release.md:17]: I doubt this is correct.

**Comment by jensens** [comment]
Forget my comments, out of scope here. Or turn them in issues, IDK.

**Comment by stevepiercy** [comment]
> Forget my comments, out of scope here. Or turn them in issues, IDK.

Heh, np. The `contribute` section needs to be reviewed to ensure it still works, especially as noted here: https://github.com/plone/plone.api/blame/b9c4fe7ee82b350da0c9b014dee228e995e7f5b9/docs/contribute/develop.md#L80-L83

**Comment by stevepiercy** [comment]
@jensens I found there is an open issue from 2017 to update the contribution docs: https://github.com/plone/plone.api/issues/357. I added it to the [Release Plone 6 Docs project board](https://github.com/orgs/plone/projects/12) in hopes that someone will take it on.

**Review by jensens — APPROVED** [review]

**Review by ksuess — CHANGES_REQUESTED** [review]
Thank you for language and grammar fixes. But please wait until I have brougth back the doctests. Thanks.

**Comment by stevepiercy** [comment]
Oh, I see what's going on now. Apparently one needs to include the preceding Python code block within the `eval-rst` block for the doctest to be properly run.

````
```{eval-rst}
..  code-block:: python

    from plone import api

    portal = api.portal.get()
    catalog = api.portal.get_tool(name="portal_catalog")
    user = api.user.create(email='alice@plone.org')

.. ifconfig:: plone_api_doctests
.. invisible-code-block: python

    self.assertEqual(portal.__class__.__name__, 'PloneSite')
    self.assertEqual(catalog.__class__.__name__, 'CatalogTool')
    self.assertEqual(user.__class__.__name__, 'MemberData')
```
````

This also results in discovering tests, not just invisible code blocks.

**Comment by stevepiercy** [comment]
I have gone as far as I can go. Here's the result of running `bin/zope-testrunner --test-path src/ -m test_doctests`.

```
  Ran 172 tests with 2 failures, 0 errors and 0 skipped in 12.372 seconds.
```

I tried to figure out the two failing tests, but nothing jumped out at me. I marked them with a `TODO:` comment. And actually there are three failing tests. There are two in `content.md` that fail, but when I comment out no tests, I only get one failing test. I tried to isolate which `assertGreater` was failing by un/commenting them. I would comment out the one test that I thought was failing, only to find that another test would fail. It's as if there is some side effect to running the tests in a specific order.

Anyway, is the count of the tests correct?

**Comment by stevepiercy** [comment]
> Thank you for language and grammar fixes. But please wait until I have brougth back the doctests. Thanks.

@ksuess I brought them back as far as I could go in this PR.

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

File: docs/contribute/develop.md (hunk 1 above)
Comment (stevepiercy) [reply docs/contribute/develop.md]: This was fixed on `master`.

File: docs/contribute/release.md (hunk 3 above)
Comment (stevepiercy) [reply docs/contribute/release.md:17]: What would be correct?

**Comment by stevepiercy** [comment]
Is there a way to tell the test runner to exclude this code block, other than removing the syntax indicator? It would be nice to have code syntax highlighting.

https://github.com/plone/plone.api/pull/476/commits/d47d85012a57e205f6b01a21414e82b6f8c2e1cc

**Comment by stevepiercy** [comment]
@jenkins-plone-org please run jobs

File: docs/contribute/release.md (hunk 3 above)
Comment (mauritsvanrees) [reply docs/contribute/release.md:17]: We use the `fullrelease` command from `zest.releaser`.
Really there is nothing special about releasing this package. I would say we completely remove the `release.md` file. It is outdated, seeming from a time when releasing packages to PyPI was new and needed to be explained.

**Comment by stevepiercy** [comment]
@ksuess or @jensens would you please review this PR, now that we have doctests? Thank you!

**Review by jensens — APPROVED** [review]

**Comment by jensens** [comment]
I would say, go for merge, but lets not forget to strip the development chapter later on.

File: docs/content.md [hunk 4]
```diff
@@ -43,7 +42,7 @@ self.assertEqual(obj.id, 'my-content')
 There are several approaches to getting your content object.
 Consider the following portal structure:
 
-```
+```console
 plone (portal root)
 |-- blog
 |-- about
```
Comment (ksuess) [inline docs/content.md:48]: A little bit nicer edge: https://github.com/plone/training/blob/main/docs/mastering-plone/installation.md#bootstrapping-a-project

File: docs/index.md [hunk 5]
```diff
@@ -28,36 +28,29 @@ Backward-incompatible changes to the API will be restricted to major versions (1
...
 :maxdepth: 1
 
-api/index.rst
-api/portal.rst
-api/content.rst
-api/user.rst
-api/group.rst
-api/env.rst
-api/relation.rst
-api/exceptions.rst
```
Comment (ksuess) [inline docs/index.md:52]: Can we keep this list of short links to API modules, please?

**Review by ksuess — APPROVED** [review]
LGTM

**Comment by stevepiercy** [comment]
> I would say, go for merge, but lets not forget to strip the development chapter later on.

@jensens @mauritsvanrees I could delete the file `docs/contribute/release.md` and modify https://6.dev-docs.plone.org/plone.api/contribute/index.html#releasing-a-new-version by deleting the section "Releasing a new version". Do you want me to purge all of that from the docs?

Alternatively I could reuse and adapt this:
https://github.com/Pylons/pylons-sphinx-themes/blob/main/RELEASE.md

I think it is important to provide instructions somewhere for how to release. Please let me know.

**Comment by mauritsvanrees** [comment]
I would say remove it.
We could use a page like this, but then in the general Plone docs in a section about contributing, not specific to plone.api.

**Comment by stevepiercy** [comment]
> I would say remove it. We could use a page like this, but then in the general Plone docs in a section about contributing, not specific to plone.api.

Created new issue in documentation to track this:

https://github.com/plone/documentation/issues/1258

Removed in e4f0ac908719dba94bcb0600ec8b0f733838b631

**Review by ksuess — APPROVED** [review]

**Comment by stevepiercy** [comment]
All the Jenkins builds fail. The failing tests are not consistent across the builds. I have no clue. I restarted the builds to see if it was just a temporary thing.

**Comment by mauritsvanrees** [comment]
> Perhaps the Jenkins failures are related to this issue? [plone/Products.CMFPlone#3542](https://github.com/plone/Products.CMFPlone/issues/3542)
> 
> If they continue to fail, is it OK for me to merge this PR?

Yes, it is the same problem. Ga ahead and merge.

**Review by mauritsvanrees — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
I have released `plone.api` 2.0.0a3 with all documentation fixes.

**Pull request merged**